projects/ngrx.io/content/guide/eslint-plugin/rules/avoid-mapping-component-store-selectors.md

```diff
@@ -0,0 +1,35 @@
+# avoid-mapping-component-store-selectors
+
+Avoid mapping logic outside the selector level.
+
+- **Type**: problem
+- **Recommended**: Yes
+- **Fixable**: No
+- **Suggestion**: No
+- **Requires type checking**: No
+- **Configurable**: No
+
+<!-- Everything above this generated, do not edit -->
+<!-- MANUAL-DOC:START -->
+
+## Rule Details
+
+Examples of **incorrect** code for this rule:
+
+```ts
+export class UserStore extends ComponentStore<UserState> {
+  loggedInUser$ = this.select((state) => state.loggedInUser);
+  //                                           ⚠ Avoid mapping logic outside the selector level.
+  name$ = this.select(this.loggedInUser$).pipe(map((user) => user.name));
```

Comment:
passing observable without a projector to the `select` method will throw a compilation error, so maybe this example would be better:

```suggestion
  name$ = this.select((state) => state.loggedInUser).pipe(map((user) => user.name));
```

Reply:
Thanks Marko!